src/keyboardAction.js

```diff
@@ -189,39 +230,35 @@ export function dndzone(node, options) {
                 }
                 break;
             }
-            case "ArrowDown":
+            case "ArrowDown": {
+                if (config.axis === "both" || config.axis === "y") {
+                    handleMoveItemUp(e);
+                } else {
+                    alertCantMoveBecauseAxis(config, "x");
+                }
+                break;
+            }
             case "ArrowRight": {
-                if (!isDragging) return;
-                e.preventDefault(); // prevent scrolling
-                e.stopPropagation();
-                const {items} = dzToConfig.get(node);
-                const children = Array.from(node.children);
-                const idx = children.indexOf(e.currentTarget);
-                printDebug(() => ["arrow down", idx]);
-                if (idx < children.length - 1) {
-                    if (!config.autoAriaDisabled) {
-                        alertToScreenReader(`Moved item ${focusedItemLabel} to position ${idx + 2} in the list ${focusedDzLabel}`);
-                    }
-                    swap(items, idx, idx + 1);
-                    dispatchFinalizeEvent(node, items, {trigger: TRIGGERS.DROPPED_INTO_ZONE, id: focusedItemId, source: SOURCES.KEYBOARD});
+                if (config.axis === "both" || config.axis === "x") {
```

Comment:
as you can see in the original code, arrow right is exactly like arrow down, it moves the item from index i to index i+1 if possible, without concerning itself with where the items are placed in space.
To achieve the axis limitation might be tricky.
It can potentially be achieved by checking bounding client rects relative to the original position of the dragged item but i think even that won't work in some edge cases

Reply:
I see. Perhaps it makes sense to only have the limit work for pointer drag, since its mainly useful for mobile devices

Reply:
Let's take a step back for a minute. What is the problem that limiting the drag axis is trying to solve from a UX perspective?

Reply:
At least for the use cases I've seen, there are often times where you're only using drag and drop for reorderability. Here's the example that led me to make this PR:
![image](https://github.com/user-attachments/assets/8cfaa5d5-01d7-46a6-a4bb-772014fbbc3e)
I have a music player, and both the queue and playlist views have a reorderable list of songs. I want it to be clear that there's no purpose to dragging other than changing the order, so limiting it to only the y-axis indicates that. 

It's also nice on mobile devices in the case where your fingers may move slightly and it keeps the element in line with the list its in

Reply:
Just thinking out loud here... If the feature was to only "allow dragging" in a direction that has other items you could displace, would that achieve the same goal? In the image you included the user can still drag upwards and leave the drop zone, with this implementation they will be able to drag up until the items becomes the first in the list and then moving the mouse further up (or to the side) won't take it further up (same way that locking to the axis doesn't let it move to the sides)

Reply:
Yea that would effectively achieve the same UX. I don't see a use case for one and not the other. Would be more work to implement tho I think

Reply:
Yeah it would be harder to implement but would be in line with the lib supporting any special arrangement of items as it does now (e.g. grid, board etc).
If you're keen I am happy to brainstorm, otherwise I can take a crack at it but not sure when I will have time (maybe over the weekend). There will need to be a util that tracks positions of items or something like that

Reply:
I'm happy to take a crack at it, there may be a relatively straightforward solution by using the indices of the items. I'm also interested in potentially converting the library to TS to make editing/maintenance easier but that would cause major conflicts with open PRs

Reply:
I like TS and there were attempts at this in the past but because of all the dom manipulation it always ended up as not adding much safety.